rca/shortcourses/tests/test_access_planit.py

```diff
@@ -0,0 +1,261 @@
+                    "https://rca.accessplanit.com/accessplansand"
+                    "box/clientinput/company/contactcompany.aspx?contacttype=8"
+                    "&coursecalid=12071"
+                ),
+            },
+        ]
+        query = QueryDict(mutable=True)
+        query.update(
+            {
+                "CompanyID": "ROYALC9RCH",
+                "courseIDs": 1,
+                "venueIDs": "",
+                "categoryIDs": "",
+            }
+        )
+        self.query = query.urlencode()
+
+    def test_xml_fetch(self):
+        """ Test the XML fetch responds. """
+        response = requests.get(
+            settings.ACCESS_PLANIT_XML_BASE_URL + self.query, timeout=5
+        )
+        self.assertEqual(response.status_code, 200)
+
+    def test_xml_fetch_no_venue(self):
+        """ Prove that you must pass blank values as parameters"""
+        query = QueryDict(mutable=True)
+        query.update({"CompanyID": "ROYALC9RCH", "courseIDs": 1})
+        query = query.urlencode()
+        response = requests.get(settings.ACCESS_PLANIT_XML_BASE_URL + query, timeout=5)
+        self.assertEqual(response.text, """Missing parameter: venueIDs.\r\n""")
+        self.assertEqual(response.status_code, 500)
+
+    """ Patch the request module totally to force the timeout so we can test the
+        result of the try/expect.
+
+        Test here that data returned from a timeout is None
+    """
+
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_data_if_timeout(self, mock_get):
+        """ If a timeout is caught the xml_data should be None"""
+        logging.disable(logging.CRITICAL)
+        mock_get.side_effect = Timeout
+        data = AccessPlanitXML(course_id=1)
+        xml_data = data.get_data()
+        self.assertEqual(xml_data, None)
+
+    @mock.patch(
+        "rca.shortcourses.access_planit.AccessPlanitXML.fetch_data_from_xml",
+        side_effect=mocked_fetch_data_from_xml,
+    )
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_cached_data_if_timeout(self, mock_get, mocked_fetch_data_from_xml):
+        """ Test getting stale cache data with a Timeout failure to get data"""
+        # logging.disable(logging.CRITICAL)
+        ShortCoursePage.objects.create(
+            title=f"Short course 1", path="1", depth="001", access_planit_course_id=1
+        )
+        # Generate the mocked xml data
+        AccessPlanitXML(course_id=1).get_data()
+        cache_key = f"short_course_1"
+        self.assertEqual(cache.get(cache_key), self.expected_data)
+        # Add a timeout for the new data request
+        mock_get.side_effect = Timeout
+        AccessPlanitXML(course_id=1).get_data()
+
+        # Now check the stale cache data is there
+        self.assertEqual(cache.get(cache_key), self.expected_data)
+
+    def test_required_course_id(self):
+        """ The course id is a required field, however if this is changed we want some tests to fail,
```

Comment:
If what is changed, the field being required or the ID itself?

Reply:
I'll fix this comment, it's a bit vague